sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -307,6 +305,14 @@ def __init__(
         container_image.capabilities.append(capability)
       pool.sdkHarnessContainerImages.append(container_image)
 
+    if not _use_fnapi(options):
+      pool.workerHarnessContainerImage = (
+          get_container_image_from_options(options))
+    elif len(pool.sdkHarnessContainerImages) == 1:
```

Comment:
I'm not super familiar with this flow - could you explain what we're trying to accomplish here/why we don't need this set in the fn_api case?

Reply:
Before cross-language, pool used to have a single `workerHarnessContainerImage` setting. With cross-langauge, this was updated to have a repeated sdkHarnessContainerImages option (so one can use different containers per langauge), but Dataflow still consults the old workerHarnessContainerImage flag in the non-cross-langauge case.

Reply:
Thanks - SGTM